sources/model/managars/GGTrackerManager.m

```diff
@@ -32,85 +32,50 @@
 #define BTTokenKey @"token"
 #define BTRatingKey @"rating"
 
+@interface GGTrackerManager()<NetworkClientUpdateDelegate>
+
+@end
+
 
 @implementation GGTrackerManager
 
 @synthesize liveMonitor = _liveMonitor;
 @synthesize appCustomer = _appCustomer;
 
 
-
-
-+ (id)tracker{
-    
-    return [self trackerWithCustomerToken:nil andDeveloperToken:nil andDelegate:nil andHTTPManager:nil];
-    
-}
-
-+ (id)trackerWithCustomerToken:(NSString *)customerToken andDeveloperToken:(NSString *)devToken andDelegate:(id <RealTimeDelegate>)delegate andHTTPManager:(GGHTTPClientManager * _Nullable)httpManager{
+- (nonnull instancetype)initWithDeveloperToken:(NSString *_Nullable)devToken HTTPManager:(GGHTTPClientManager * _Nullable)httpManager realTimeDelegate:(id <RealTimeDelegate> _Nullable)realTimeDelegate{
  
-    static GGTrackerManager *sharedObject = nil;
-    static dispatch_once_t onceToken;
-    dispatch_once(&onceToken, ^{
-        
-        // init the tracker
-        sharedObject = [[self alloc] initTacker];
-        
+    
+     if (self = [super init]) {
+   
         // init the real time monitor
-        sharedObject->_liveMonitor = [GGRealTimeMontior sharedInstance];
-        sharedObject->_liveMonitor.realtimeConnectionDelegate = sharedObject;
+        self.liveMonitor = [[GGRealTimeMontior alloc] init];
+        self.liveMonitor.realtimeConnectionDelegate = self;
         
         // init polled
-        sharedObject->_polledOrders = [NSMutableSet set];
-        sharedObject->_polledLocations = [NSMutableSet set];
+        self.polledOrders = [NSMutableSet set];
+        self.polledLocations = [NSMutableSet set];
         
         // setup http manager
-        sharedObject->_httpManager = httpManager;
+        self.httpManager = httpManager;
         
-        sharedObject->_shouldReconnect = YES;
+        self.shouldReconnect = YES;
         
-        sharedObject->_numConnectionAttempts = 0;
+        self.numConnectionAttempts = 0;
         
-        // configure observers
-        [sharedObject configureObservers];
-    });
-    
-    // set the customer token and developer token
-    if (customerToken) [sharedObject setCustomerToken:customerToken];
-    if (devToken) [sharedObject setDeveloperToken:devToken];
-    
-    // set the connection delegate
-    if (delegate) [sharedObject setRealTimeDelegate:delegate];
-    
-    return sharedObject;
-}
-
-
--(nonnull instancetype)initTacker{
-    if (self = [super init]) {
-        self.logsEnabled = NO;
-    }
+        // set the customer token and developer token
+        if (devToken) [self setDeveloperToken:devToken];
+        
+        // set the connection delegate
+        if (realTimeDelegate) [self setRealTimeDelegate:realTimeDelegate];
```

Comment:
what do you mean? if i get nil in the realtime delegate  i don't do any thing with it